pkg/inject/report.go

```diff
@@ -125,12 +132,29 @@ func (r *Report) Injectable() (bool, []string) {
 		reasons = append(reasons, disabledAutomountServiceAccountToken)
 	}
 
+	if !r.ServiceAccountAutomountToken {
+		reasons = append(reasons, disabledServiceAccountAutomountToken)
+	}
+
 	if len(reasons) > 0 {
 		return false, reasons
 	}
 	return true, nil
 }
 
+func checkServiceAccountVolumeMount(t *v1.PodSpec) bool {
```

Comment:
To remove code duplication, can you replace calls to `ResourceConfig.serviceAccountVolumeMount` with calls to this function instead?